components/ws-manager-bridge/src/prebuild-state-mapper.ts

```diff
@@ -16,74 +18,81 @@ export interface PrebuildUpdate {
 
 @injectable()
 export class PrebuildStateMapper {
-    mapWorkspaceStatusToPrebuild(status: WorkspaceStatus.AsObject): PrebuildUpdate | undefined {
-        if (status.phase === WorkspacePhase.STOPPED) {
-            // Ideally, we'd love to hande STOPPED identical to STOPPING, because we want to assume that all conditions are stable.
-            // For reliabilies sake we don't do it because experiences shows that unstable conditions are one of the most common sources of errors.
+    constructor(@inject(ExperimentsClient) private readonly experimentsClient: ExperimentsClient) {}
+    async mapWorkspaceStatusToPrebuild(status: WorkspaceStatus.AsObject): Promise<PrebuildUpdate | undefined> {
+        const canUseStoppedPhase = await this.experimentsClient.getValueAsync(
+            "ws_manager_bridge_stopped_prebuild_statuses",
+            false,
+            {},
+        );
+        if (!canUseStoppedPhase && status.phase === WorkspacePhase.STOPPED) {
```

Comment:
IMO the following would be easier to understand:
```
If (!canUseStoppedPhase) {
    if (STOPPED) {
        return undefined
    }
    if (!STOPPING) {
        return "building"
    }
} else {
    if (!STOPPED) {
        return "building"
    }
}
// both cases fall-through to the state deduction
```

Leaving it up to you whether you want to change or not (both are equivalent).